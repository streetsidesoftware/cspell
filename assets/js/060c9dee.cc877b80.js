"use strict";(self.webpackChunkwebsite=self.webpackChunkwebsite||[]).push([[94023],{46039:(e,t,n)=>{n.d(t,{R:()=>l,x:()=>c});var i=n(18225);const s={},r=i.createContext(s);function l(e){const t=i.useContext(r);return i.useMemo((function(){return"function"==typeof e?e(t):{...t,...e}}),[t,e])}function c(e){let t;return t=e.disableParentContext?"function"==typeof e.components?e.components(s):e.components||s:l(e.components),i.createElement(r.Provider,{value:t},e.children)}},87257:(e,t,n)=>{n.r(t),n.d(t,{assets:()=>a,contentTitle:()=>c,default:()=>p,frontMatter:()=>l,metadata:()=>i,toc:()=>o});const i=JSON.parse('{"id":"api/cspell-lib/functions/readRawSettings","title":"Function: readRawSettings()","description":"readRawSettings(filename, relativeTo?): Promise\\\\","source":"@site/docs/api/cspell-lib/functions/readRawSettings.md","sourceDirName":"api/cspell-lib/functions","slug":"/api/cspell-lib/functions/readRawSettings","permalink":"/docs/api/cspell-lib/functions/readRawSettings","draft":false,"unlisted":false,"editUrl":"https://github.com/streetsidesoftware/cspell/tree/main/website/docs/docs/api/cspell-lib/functions/readRawSettings.md","tags":[],"version":"current","frontMatter":{},"sidebar":"tutorialSidebar","previous":{"title":"Function: readFileSync()","permalink":"/docs/api/cspell-lib/functions/readFileSync"},"next":{"title":"Function: readSettings()","permalink":"/docs/api/cspell-lib/functions/readSettings"}}');var s=n(37557),r=n(46039);const l={},c="Function: readRawSettings()",a={},o=[{value:"Parameters",id:"parameters",level:2},{value:"filename",id:"filename",level:3},{value:"relativeTo?",id:"relativeto",level:3},{value:"Returns",id:"returns",level:2}];function d(e){const t={a:"a",blockquote:"blockquote",code:"code",h1:"h1",h2:"h2",h3:"h3",header:"header",p:"p",strong:"strong",...(0,r.R)(),...e.components};return(0,s.jsxs)(s.Fragment,{children:[(0,s.jsx)(t.header,{children:(0,s.jsx)(t.h1,{id:"function-readrawsettings",children:"Function: readRawSettings()"})}),"\n",(0,s.jsxs)(t.blockquote,{children:["\n",(0,s.jsxs)(t.p,{children:[(0,s.jsx)(t.strong,{children:"readRawSettings"}),"(",(0,s.jsx)(t.code,{children:"filename"}),", ",(0,s.jsx)(t.code,{children:"relativeTo"}),"?): ",(0,s.jsx)(t.code,{children:"Promise"}),"<",(0,s.jsx)(t.a,{href:"/docs/api/cspell-lib/interfaces/CSpellSettingsWithSourceTrace",children:(0,s.jsx)(t.code,{children:"CSpellSettingsWithSourceTrace"})}),">"]}),"\n"]}),"\n",(0,s.jsxs)(t.p,{children:["Defined in: ",(0,s.jsx)(t.a,{href:"https://github.com/streetsidesoftware/cspell/blob/3fc3ef2ea9def847ef20287c6238ff423e6493eb/packages/cspell-lib/src/lib/Settings/Controller/configLoader/defaultConfigLoader.ts#L99",children:"packages/cspell-lib/src/lib/Settings/Controller/configLoader/defaultConfigLoader.ts:99"})]}),"\n",(0,s.jsx)(t.h2,{id:"parameters",children:"Parameters"}),"\n",(0,s.jsx)(t.h3,{id:"filename",children:"filename"}),"\n",(0,s.jsxs)(t.p,{children:[(0,s.jsx)(t.code,{children:"string"})," | ",(0,s.jsx)(t.code,{children:"URL"})]}),"\n",(0,s.jsx)(t.h3,{id:"relativeto",children:"relativeTo?"}),"\n",(0,s.jsxs)(t.p,{children:[(0,s.jsx)(t.code,{children:"string"})," | ",(0,s.jsx)(t.code,{children:"URL"})]}),"\n",(0,s.jsx)(t.h2,{id:"returns",children:"Returns"}),"\n",(0,s.jsxs)(t.p,{children:[(0,s.jsx)(t.code,{children:"Promise"}),"<",(0,s.jsx)(t.a,{href:"/docs/api/cspell-lib/interfaces/CSpellSettingsWithSourceTrace",children:(0,s.jsx)(t.code,{children:"CSpellSettingsWithSourceTrace"})}),">"]})]})}function p(e={}){const{wrapper:t}={...(0,r.R)(),...e.components};return t?(0,s.jsx)(t,{...e,children:(0,s.jsx)(d,{...e})}):d(e)}}}]);