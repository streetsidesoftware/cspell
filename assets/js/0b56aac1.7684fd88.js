"use strict";(self.webpackChunkwebsite=self.webpackChunkwebsite||[]).push([[16718],{17322:(e,s,i)=>{i.r(s),i.d(s,{assets:()=>o,contentTitle:()=>a,default:()=>p,frontMatter:()=>r,metadata:()=>l,toc:()=>c});const l=JSON.parse('{"id":"api/cspell-lib/variables/ConfigFields","title":"Variable: ConfigFields","description":"const ConfigFields: CSpellUserSettingsFields","source":"@site/docs/api/cspell-lib/variables/ConfigFields.md","sourceDirName":"api/cspell-lib/variables","slug":"/api/cspell-lib/variables/ConfigFields","permalink":"/docs/api/cspell-lib/variables/ConfigFields","draft":false,"unlisted":false,"editUrl":"https://github.com/streetsidesoftware/cspell/tree/main/website/docs/docs/api/cspell-lib/variables/ConfigFields.md","tags":[],"version":"current","frontMatter":{},"sidebar":"tutorialSidebar","previous":{"title":"Type Alias: ~~VersionLegacy~~","permalink":"/docs/api/cspell-lib/type-aliases/VersionLegacy"},"next":{"title":"Variable: ENV\\\\_CSPELL\\\\_GLOB\\\\_ROOT","permalink":"/docs/api/cspell-lib/variables/ENV_CSPELL_GLOB_ROOT"}}');var t=i(37557),n=i(46039);const r={},a="Variable: ConfigFields",o={},c=[];function d(e){const s={a:"a",blockquote:"blockquote",code:"code",h1:"h1",header:"header",p:"p",strong:"strong",...(0,n.R)(),...e.components};return(0,t.jsxs)(t.Fragment,{children:[(0,t.jsx)(s.header,{children:(0,t.jsx)(s.h1,{id:"variable-configfields",children:"Variable: ConfigFields"})}),"\n",(0,t.jsxs)(s.blockquote,{children:["\n",(0,t.jsxs)(s.p,{children:[(0,t.jsx)(s.code,{children:"const"})," ",(0,t.jsx)(s.strong,{children:"ConfigFields"}),": ",(0,t.jsx)(s.a,{href:"/docs/api/cspell-lib/type-aliases/CSpellUserSettingsFields",children:(0,t.jsx)(s.code,{children:"CSpellUserSettingsFields"})})]}),"\n"]}),"\n",(0,t.jsx)(s.p,{children:"Defined in: packages/cspell-types/dist/index.d.mts:1566"})]})}function p(e={}){const{wrapper:s}={...(0,n.R)(),...e.components};return s?(0,t.jsx)(s,{...e,children:(0,t.jsx)(d,{...e})}):d(e)}},46039:(e,s,i)=>{i.d(s,{R:()=>r,x:()=>a});var l=i(18225);const t={},n=l.createContext(t);function r(e){const s=l.useContext(n);return l.useMemo((function(){return"function"==typeof e?e(s):{...s,...e}}),[s,e])}function a(e){let s;return s=e.disableParentContext?"function"==typeof e.components?e.components(t):e.components||t:r(e.components),l.createElement(n.Provider,{value:s},e.children)}}}]);