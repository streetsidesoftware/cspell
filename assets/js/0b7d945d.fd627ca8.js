"use strict";(self.webpackChunkwebsite=self.webpackChunkwebsite||[]).push([[43752],{46039:(e,s,t)=>{t.d(s,{R:()=>o,x:()=>i});var n=t(18225);const r={},c=n.createContext(r);function o(e){const s=n.useContext(c);return n.useMemo((function(){return"function"==typeof e?e(s):{...s,...e}}),[s,e])}function i(e){let s;return s=e.disableParentContext?"function"==typeof e.components?e.components(r):e.components||r:o(e.components),n.createElement(c.Provider,{value:s},e.children)}},69483:(e,s,t)=>{t.r(s),t.d(s,{assets:()=>l,contentTitle:()=>i,default:()=>p,frontMatter:()=>o,metadata:()=>n,toc:()=>a});const n=JSON.parse('{"id":"api/cspell/functions/trace","title":"Function: trace()","description":"trace(words, options): AsyncIterableIterator\\\\","source":"@site/docs/api/cspell/functions/trace.md","sourceDirName":"api/cspell/functions","slug":"/api/cspell/functions/trace","permalink":"/docs/api/cspell/functions/trace","draft":false,"unlisted":false,"editUrl":"https://github.com/streetsidesoftware/cspell/tree/main/website/docs/docs/api/cspell/functions/trace.md","tags":[],"version":"current","frontMatter":{},"sidebar":"tutorialSidebar","previous":{"title":"Function: suggestions()","permalink":"/docs/api/cspell/functions/suggestions"},"next":{"title":"cspell","permalink":"/docs/api/cspell/globals"}}');var r=t(37557),c=t(46039);const o={},i="Function: trace()",l={},a=[{value:"Parameters",id:"parameters",level:2},{value:"words",id:"words",level:3},{value:"options",id:"options",level:3},{value:"Returns",id:"returns",level:2}];function d(e){const s={a:"a",blockquote:"blockquote",code:"code",h1:"h1",h2:"h2",h3:"h3",header:"header",p:"p",strong:"strong",...(0,c.R)(),...e.components};return(0,r.jsxs)(r.Fragment,{children:[(0,r.jsx)(s.header,{children:(0,r.jsx)(s.h1,{id:"function-trace",children:"Function: trace()"})}),"\n",(0,r.jsxs)(s.blockquote,{children:["\n",(0,r.jsxs)(s.p,{children:[(0,r.jsx)(s.strong,{children:"trace"}),"(",(0,r.jsx)(s.code,{children:"words"}),", ",(0,r.jsx)(s.code,{children:"options"}),"): ",(0,r.jsx)(s.code,{children:"AsyncIterableIterator"}),"<",(0,r.jsx)(s.code,{children:"TraceWordResult"}),">"]}),"\n"]}),"\n",(0,r.jsxs)(s.p,{children:["Defined in: ",(0,r.jsx)(s.a,{href:"https://github.com/streetsidesoftware/cspell/blob/3fc3ef2ea9def847ef20287c6238ff423e6493eb/packages/cspell/src/app/application.mts#L44",children:"cspell/src/app/application.mts:44"})]}),"\n",(0,r.jsx)(s.h2,{id:"parameters",children:"Parameters"}),"\n",(0,r.jsx)(s.h3,{id:"words",children:"words"}),"\n",(0,r.jsxs)(s.p,{children:[(0,r.jsx)(s.code,{children:"string"}),"[]"]}),"\n",(0,r.jsx)(s.h3,{id:"options",children:"options"}),"\n",(0,r.jsx)(s.p,{children:(0,r.jsx)(s.a,{href:"/docs/api/cspell/interfaces/TraceOptions",children:(0,r.jsx)(s.code,{children:"TraceOptions"})})}),"\n",(0,r.jsx)(s.h2,{id:"returns",children:"Returns"}),"\n",(0,r.jsxs)(s.p,{children:[(0,r.jsx)(s.code,{children:"AsyncIterableIterator"}),"<",(0,r.jsx)(s.code,{children:"TraceWordResult"}),">"]})]})}function p(e={}){const{wrapper:s}={...(0,c.R)(),...e.components};return s?(0,r.jsx)(s,{...e,children:(0,r.jsx)(d,{...e})}):d(e)}}}]);