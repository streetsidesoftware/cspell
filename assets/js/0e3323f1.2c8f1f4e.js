"use strict";(self.webpackChunkwebsite=self.webpackChunkwebsite||[]).push([[25350],{4002:(e,s,n)=>{n.r(s),n.d(s,{assets:()=>c,contentTitle:()=>o,default:()=>u,frontMatter:()=>r,metadata:()=>t,toc:()=>a});const t=JSON.parse('{"id":"api/cspell-types/interfaces/SuggestionsConfiguration","title":"Interface: SuggestionsConfiguration","description":"Defined in286","source":"@site/docs/api/cspell-types/interfaces/SuggestionsConfiguration.md","sourceDirName":"api/cspell-types/interfaces","slug":"/api/cspell-types/interfaces/SuggestionsConfiguration","permalink":"/docs/api/cspell-types/interfaces/SuggestionsConfiguration","draft":false,"unlisted":false,"editUrl":"https://github.com/streetsidesoftware/cspell/tree/main/website/docs/docs/api/cspell-types/interfaces/SuggestionsConfiguration.md","tags":[],"version":"current","frontMatter":{},"sidebar":"tutorialSidebar","previous":{"title":"Interface: Settings","permalink":"/docs/api/cspell-types/interfaces/Settings"},"next":{"title":"Interface: TextDocumentOffset","permalink":"/docs/api/cspell-types/interfaces/TextDocumentOffset"}}');var i=n(37557),l=n(46039);const r={},o="Interface: SuggestionsConfiguration",c={},a=[{value:"Extended by",id:"extended-by",level:2},{value:"Properties",id:"properties",level:2},{value:"numSuggestions?",id:"numsuggestions",level:3},{value:"Default",id:"default",level:4},{value:"suggestionNumChanges?",id:"suggestionnumchanges",level:3},{value:"Default",id:"default-1",level:4},{value:"suggestionsTimeout?",id:"suggestionstimeout",level:3},{value:"Default",id:"default-2",level:4}];function d(e){const s={a:"a",blockquote:"blockquote",code:"code",h1:"h1",h2:"h2",h3:"h3",h4:"h4",header:"header",hr:"hr",li:"li",p:"p",pre:"pre",strong:"strong",ul:"ul",...(0,l.R)(),...e.components};return(0,i.jsxs)(i.Fragment,{children:[(0,i.jsx)(s.header,{children:(0,i.jsx)(s.h1,{id:"interface-suggestionsconfiguration",children:"Interface: SuggestionsConfiguration"})}),"\n",(0,i.jsxs)(s.p,{children:["Defined in: ",(0,i.jsx)(s.a,{href:"https://github.com/streetsidesoftware/cspell/blob/3fc3ef2ea9def847ef20287c6238ff423e6493eb/packages/cspell-types/src/CSpellSettingsDef.ts#L286",children:"CSpellSettingsDef.ts:286"})]}),"\n",(0,i.jsx)(s.h2,{id:"extended-by",children:"Extended by"}),"\n",(0,i.jsxs)(s.ul,{children:["\n",(0,i.jsx)(s.li,{children:(0,i.jsx)(s.a,{href:"/docs/api/cspell-types/interfaces/ReportingConfiguration",children:(0,i.jsx)(s.code,{children:"ReportingConfiguration"})})}),"\n"]}),"\n",(0,i.jsx)(s.h2,{id:"properties",children:"Properties"}),"\n",(0,i.jsx)(s.h3,{id:"numsuggestions",children:"numSuggestions?"}),"\n",(0,i.jsxs)(s.blockquote,{children:["\n",(0,i.jsxs)(s.p,{children:[(0,i.jsx)(s.code,{children:"optional"})," ",(0,i.jsx)(s.strong,{children:"numSuggestions"}),": ",(0,i.jsx)(s.code,{children:"number"})]}),"\n"]}),"\n",(0,i.jsxs)(s.p,{children:["Defined in: ",(0,i.jsx)(s.a,{href:"https://github.com/streetsidesoftware/cspell/blob/3fc3ef2ea9def847ef20287c6238ff423e6493eb/packages/cspell-types/src/CSpellSettingsDef.ts#L292",children:"CSpellSettingsDef.ts:292"})]}),"\n",(0,i.jsx)(s.p,{children:"Number of suggestions to make."}),"\n",(0,i.jsx)(s.h4,{id:"default",children:"Default"}),"\n",(0,i.jsx)(s.pre,{children:(0,i.jsx)(s.code,{className:"language-ts",children:"10\n"})}),"\n",(0,i.jsx)(s.hr,{}),"\n",(0,i.jsx)(s.h3,{id:"suggestionnumchanges",children:"suggestionNumChanges?"}),"\n",(0,i.jsxs)(s.blockquote,{children:["\n",(0,i.jsxs)(s.p,{children:[(0,i.jsx)(s.code,{children:"optional"})," ",(0,i.jsx)(s.strong,{children:"suggestionNumChanges"}),": ",(0,i.jsx)(s.code,{children:"number"})]}),"\n"]}),"\n",(0,i.jsxs)(s.p,{children:["Defined in: ",(0,i.jsx)(s.a,{href:"https://github.com/streetsidesoftware/cspell/blob/3fc3ef2ea9def847ef20287c6238ff423e6493eb/packages/cspell-types/src/CSpellSettingsDef.ts#L310",children:"CSpellSettingsDef.ts:310"})]}),"\n",(0,i.jsx)(s.p,{children:"The maximum number of changes allowed on a word to be considered a suggestions."}),"\n",(0,i.jsxs)(s.p,{children:["For example, appending an ",(0,i.jsx)(s.code,{children:"s"})," onto ",(0,i.jsx)(s.code,{children:"example"})," -> ",(0,i.jsx)(s.code,{children:"examples"})," is considered 1 change."]}),"\n",(0,i.jsx)(s.p,{children:"Range: between 1 and 5."}),"\n",(0,i.jsx)(s.h4,{id:"default-1",children:"Default"}),"\n",(0,i.jsx)(s.pre,{children:(0,i.jsx)(s.code,{className:"language-ts",children:"3\n"})}),"\n",(0,i.jsx)(s.hr,{}),"\n",(0,i.jsx)(s.h3,{id:"suggestionstimeout",children:"suggestionsTimeout?"}),"\n",(0,i.jsxs)(s.blockquote,{children:["\n",(0,i.jsxs)(s.p,{children:[(0,i.jsx)(s.code,{children:"optional"})," ",(0,i.jsx)(s.strong,{children:"suggestionsTimeout"}),": ",(0,i.jsx)(s.code,{children:"number"})]}),"\n"]}),"\n",(0,i.jsxs)(s.p,{children:["Defined in: ",(0,i.jsx)(s.a,{href:"https://github.com/streetsidesoftware/cspell/blob/3fc3ef2ea9def847ef20287c6238ff423e6493eb/packages/cspell-types/src/CSpellSettingsDef.ts#L299",children:"CSpellSettingsDef.ts:299"})]}),"\n",(0,i.jsx)(s.p,{children:"The maximum amount of time in milliseconds to generate suggestions for a word."}),"\n",(0,i.jsx)(s.h4,{id:"default-2",children:"Default"}),"\n",(0,i.jsx)(s.pre,{children:(0,i.jsx)(s.code,{className:"language-ts",children:"500\n"})})]})}function u(e={}){const{wrapper:s}={...(0,l.R)(),...e.components};return s?(0,i.jsx)(s,{...e,children:(0,i.jsx)(d,{...e})}):d(e)}},46039:(e,s,n)=>{n.d(s,{R:()=>r,x:()=>o});var t=n(18225);const i={},l=t.createContext(i);function r(e){const s=t.useContext(l);return t.useMemo((function(){return"function"==typeof e?e(s):{...s,...e}}),[s,e])}function o(e){let s;return s=e.disableParentContext?"function"==typeof e.components?e.components(i):e.components||i:r(e.components),t.createElement(l.Provider,{value:s},e.children)}}}]);