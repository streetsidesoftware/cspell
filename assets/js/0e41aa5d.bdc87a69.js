"use strict";(self.webpackChunkwebsite=self.webpackChunkwebsite||[]).push([[64458],{1949:(e,r,s)=>{s.r(r),s.d(r,{assets:()=>c,contentTitle:()=>l,default:()=>d,frontMatter:()=>o,metadata:()=>t,toc:()=>a});const t=JSON.parse('{"id":"api/cspell-lib/type-aliases/ErrorLike","title":"Type Alias: ErrorLike","description":"ErrorLike string; name () => string; \\\\}","source":"@site/docs/api/cspell-lib/type-aliases/ErrorLike.md","sourceDirName":"api/cspell-lib/type-aliases","slug":"/api/cspell-lib/type-aliases/ErrorLike","permalink":"/docs/api/cspell-lib/type-aliases/ErrorLike","draft":false,"unlisted":false,"editUrl":"https://github.com/streetsidesoftware/cspell/tree/main/website/docs/docs/api/cspell-lib/type-aliases/ErrorLike.md","tags":[],"version":"current","frontMatter":{},"sidebar":"tutorialSidebar","previous":{"title":"Type Alias: ErrorEmitter()","permalink":"/docs/api/cspell-lib/type-aliases/ErrorEmitter"},"next":{"title":"Type Alias: ExclusionFunction()","permalink":"/docs/api/cspell-lib/type-aliases/ExclusionFunction"}}');var i=s(37557),n=s(46039);const o={},l="Type Alias: ErrorLike",c={},a=[];function p(e){const r={blockquote:"blockquote",code:"code",h1:"h1",header:"header",p:"p",strong:"strong",...(0,n.R)(),...e.components};return(0,i.jsxs)(i.Fragment,{children:[(0,i.jsx)(r.header,{children:(0,i.jsx)(r.h1,{id:"type-alias-errorlike",children:"Type Alias: ErrorLike"})}),"\n",(0,i.jsxs)(r.blockquote,{children:["\n",(0,i.jsxs)(r.p,{children:[(0,i.jsx)(r.strong,{children:"ErrorLike"}),": ",(0,i.jsx)(r.code,{children:"Error"})," | { ",(0,i.jsx)(r.code,{children:"message"}),": ",(0,i.jsx)(r.code,{children:"string"}),"; ",(0,i.jsx)(r.code,{children:"name"}),": ",(0,i.jsx)(r.code,{children:"string"}),"; ",(0,i.jsx)(r.code,{children:"toString"}),": () => ",(0,i.jsx)(r.code,{children:"string"}),"; }"]}),"\n"]}),"\n",(0,i.jsx)(r.p,{children:"Defined in: packages/cspell-types/dist/index.d.mts:76"})]})}function d(e={}){const{wrapper:r}={...(0,n.R)(),...e.components};return r?(0,i.jsx)(r,{...e,children:(0,i.jsx)(p,{...e})}):p(e)}},46039:(e,r,s)=>{s.d(r,{R:()=>o,x:()=>l});var t=s(18225);const i={},n=t.createContext(i);function o(e){const r=t.useContext(n);return t.useMemo((function(){return"function"==typeof e?e(r):{...r,...e}}),[r,e])}function l(e){let r;return r=e.disableParentContext?"function"==typeof e.components?e.components(i):e.components||i:o(e.components),t.createElement(n.Provider,{value:r},e.children)}}}]);