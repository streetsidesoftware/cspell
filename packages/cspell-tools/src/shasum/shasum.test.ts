import { writeFile } from 'node:fs/promises';

import { afterEach, describe, expect, test, vi } from 'vitest';

import { resolvePathToFixture } from '../test/TestHelper.js';
import {
    calcUpdateChecksumForFiles,
    checkShasumFile,
    reportCheckChecksumFile,
    reportChecksumForFiles,
    updateChecksumForFiles,
} from './shasum.js';

vi.mock('node:fs/promises', async () => {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    const fs: any = await vi.importActual('node:fs/promises');
    return {
        ...fs,
        writeFile: vi.fn().mockImplementation(() => Promise.resolve(undefined)),
    };
});

const mockedWriteFile = vi.mocked(writeFile);

describe('shasum', () => {
    afterEach(() => {
        vi.resetAllMocks();
    });

    test('checkShasumFile pass', async () => {
        const root = resolvePathToFixture('dicts');
        const filename = resolvePathToFixture('dicts/_checksum.txt');
        const result = await checkShasumFile(filename, [], root);
        expect(result.filter((r) => !r.passed)).toHaveLength(0);
        expect(result).toMatchSnapshot();
    });

    test('checkShasumFile pass with files', async () => {
        const root = resolvePathToFixture('dicts');
        const filename = resolvePathToFixture('dicts/_checksum.txt');
        const files = ['colors.txt', 'cities.txt'];
        const result = await checkShasumFile(filename, files, root);
        expect(result.filter((r) => !r.passed)).toHaveLength(0);
        expect(result.map((r) => r.filename)).toEqual(files);
        expect(result).toMatchSnapshot();
    });

    test('checkShasumFile pass with files but file not in checksum.txt', async () => {
        const root = resolvePathToFixture('dicts');
        const filename = resolvePathToFixture('dicts/_checksum.txt');
        const files = ['colors.txt', 'my_cities.txt'];
        const result = await checkShasumFile(filename, files, root);
        expect(result.filter((r) => !r.passed)).toHaveLength(1);
        expect(result.map((r) => r.filename)).toEqual(files);
        // console.error('%o', result);
    });

    test('checkShasumFile not pass', async () => {
        const root = resolvePathToFixture('dicts');
        const filename = resolvePathToFixture('dicts/_checksum-failed.txt');
        const result = await checkShasumFile(filename, [], root);
        expect(result.filter((r) => !r.passed)).toHaveLength(1);
        expect(result).toMatchSnapshot();
    });

    test('checkShasumFile missing files', async () => {
        const root = resolvePathToFixture('dicts');
        const filename = resolvePathToFixture('dicts/_checksum-missing-file.txt');
        const result = await checkShasumFile(filename, [], root);
        // console.error('%o', result);
        expect(result.filter((r) => !r.passed)).toHaveLength(1);
        const missingResult = result[0];
        expect(missingResult.error).toEqual(Error('Failed to read file.'));
    });

    test('checkShasumFile bad format', async () => {
        const root = resolvePathToFixture('dicts');
        const filename = resolvePathToFixture('dicts/_checksum-bad-format.txt');
        await expect(checkShasumFile(filename, [], root)).rejects.toEqual(
            Error('Failed to parse line 3 of checksum file.')
        );
    });

    test('reportChecksumForFiles', async () => {
        const root = resolvePathToFixture('dicts');
        const files = ['colors.txt', 'cities.txt'];
        const report = await reportChecksumForFiles(files, { root });
        expect(report).toMatchSnapshot();
    });

    test.each`
        filename                        | files                              | listFile
        ${'_checksum.txt'}              | ${undefined}                       | ${undefined}
        ${'_checksum.txt'}              | ${['colors.txt', 'my_cities.txt']} | ${undefined}
        ${'_checksum-failed.txt'}       | ${undefined}                       | ${undefined}
        ${'_checksum-failed2.txt'}      | ${undefined}                       | ${undefined}
        ${'_checksum-failed2.txt'}      | ${['cities.txt', 'colors.txt']}    | ${undefined}
        ${'_checksum-missing-file.txt'} | ${undefined}                       | ${undefined}
        ${'_checksum.txt'}              | ${['colors.txt', 'my_cities.txt']} | ${'source-files.txt'}
    `('reportCheckChecksumFile $filename $files $listFile', async ({ filename, files, listFile }) => {
        const root = resolvePathToFixture('dicts');
        const report = await reportCheckChecksumFile(resolvePathToFixture('dicts', filename), files, {
            root,
            listFile: listFile ? [resolvePathToFixture('dicts', listFile)] : undefined,
        });
        expect(report).toMatchSnapshot();
    });

    test.each`
        filename
        ${'_checksum-failed.txt'}
        ${'_checksum-missing-file'}
        ${'new_checksum_file.txt'}
    `('calcUpdateChecksumForFiles $filename', async ({ filename }) => {
        const root = resolvePathToFixture('dicts');
        const checksumFile = resolvePathToFixture('dicts', filename);
        const listFile = resolvePathToFixture('dicts', 'source-files.txt');

        const result = await calcUpdateChecksumForFiles(checksumFile, [], { root, listFile: [listFile] });
        expect(result).toMatchSnapshot();
    });

    test('updateChecksumForFiles', async () => {
        const checksumFile = 'temp/my-checksum.txt';
        const root = resolvePathToFixture('dicts');
        const listFile = resolvePathToFixture('dicts', 'source-files.txt');

        const result = await updateChecksumForFiles(checksumFile, [], { root, listFile: [listFile] });
        expect(mockedWriteFile).toHaveBeenLastCalledWith(checksumFile, result.report);
    });
});
