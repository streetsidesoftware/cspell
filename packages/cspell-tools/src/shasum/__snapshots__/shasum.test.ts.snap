// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`shasum > calcUpdateChecksumForFiles '_checksum-failed.txt' 1`] = `
"a8ab65e8305f4911577525c8e950fcea6667de59  cities.trie
963a65138d4391c8de2f0dfb5a7ef890e512a95e  cities.trie.gz
477e8bc9954033392e432fd66f0d4278884bde38  cities.txt
55915445afc07bf877eea1e982aecb7c97b247da  colors.trie
3a6b55a089d018878e8b904f8f19391f2e30b66c  colors.txt
25a493fa62702d3e052717a26c6740a30614457e  sampleCodeDic.txt
"
`;

exports[`shasum > calcUpdateChecksumForFiles '_checksum-missing-file' 1`] = `
"477e8bc9954033392e432fd66f0d4278884bde38  cities.txt
3a6b55a089d018878e8b904f8f19391f2e30b66c  colors.txt
"
`;

exports[`shasum > calcUpdateChecksumForFiles 'new_checksum_file.txt' 1`] = `
"477e8bc9954033392e432fd66f0d4278884bde38  cities.txt
3a6b55a089d018878e8b904f8f19391f2e30b66c  colors.txt
"
`;

exports[`shasum > checkShasumFile not pass 1`] = `
[
  {
    "filename": "cities.trie",
    "passed": true,
  },
  {
    "filename": "cities.trie.gz",
    "passed": true,
  },
  {
    "filename": "cities.txt",
    "passed": true,
  },
  {
    "filename": "colors.trie",
    "passed": false,
  },
  {
    "filename": "colors.txt",
    "passed": true,
  },
  {
    "filename": "sampleCodeDic.txt",
    "passed": true,
  },
]
`;

exports[`shasum > checkShasumFile pass 1`] = `
[
  {
    "filename": "cities.trie",
    "passed": true,
  },
  {
    "filename": "cities.trie.gz",
    "passed": true,
  },
  {
    "filename": "cities.txt",
    "passed": true,
  },
  {
    "filename": "colors.trie",
    "passed": true,
  },
  {
    "filename": "colors.txt",
    "passed": true,
  },
  {
    "filename": "sampleCodeDic.txt",
    "passed": true,
  },
]
`;

exports[`shasum > checkShasumFile pass with files 1`] = `
[
  {
    "filename": "colors.txt",
    "passed": true,
  },
  {
    "filename": "cities.txt",
    "passed": true,
  },
]
`;

exports[`shasum > reportCheckChecksumFile '_checksum.txt' [ 'colors.txt', 'my_cities.txt' ] 'source-files.txt' 1`] = `
{
  "passed": false,
  "report": "colors.txt: OK
my_cities.txt: FAILED - Missing Checksum.
cities.txt: OK
shasum: WARNING: 1 computed checksum did NOT match",
}
`;

exports[`shasum > reportCheckChecksumFile '_checksum.txt' [ 'colors.txt', 'my_cities.txt' ] undefined 1`] = `
{
  "passed": false,
  "report": "colors.txt: OK
my_cities.txt: FAILED - Missing Checksum.
shasum: WARNING: 1 computed checksum did NOT match",
}
`;

exports[`shasum > reportCheckChecksumFile '_checksum.txt' undefined undefined 1`] = `
{
  "passed": true,
  "report": "cities.trie: OK
cities.trie.gz: OK
cities.txt: OK
colors.trie: OK
colors.txt: OK
sampleCodeDic.txt: OK",
}
`;

exports[`shasum > reportCheckChecksumFile '_checksum-failed.txt' undefined undefined 1`] = `
{
  "passed": false,
  "report": "cities.trie: OK
cities.trie.gz: OK
cities.txt: OK
colors.trie: FAILED
colors.txt: OK
sampleCodeDic.txt: OK
shasum: WARNING: 1 computed checksum did NOT match",
}
`;

exports[`shasum > reportCheckChecksumFile '_checksum-failed2.txt' [ 'cities.txt', 'colors.txt' ] undefined 1`] = `
{
  "passed": true,
  "report": "cities.txt: OK
colors.txt: OK",
}
`;

exports[`shasum > reportCheckChecksumFile '_checksum-failed2.txt' undefined undefined 1`] = `
{
  "passed": false,
  "report": "cities.trie: OK
cities.trie.gz: OK
cities.txt: OK
colors.trie: FAILED
colors.txt: OK
sampleCodeDic.txt: OK
missing-file.txt: FAILED - Failed to read file.
shasum: WARNING: 2 computed checksums did NOT match",
}
`;

exports[`shasum > reportCheckChecksumFile '_checksum-missing-file.txt' undefined undefined 1`] = `
{
  "passed": false,
  "report": "missing-file.txt: FAILED - Failed to read file.
shasum: WARNING: 1 computed checksum did NOT match",
}
`;

exports[`shasum > reportChecksumForFiles 1`] = `
{
  "passed": true,
  "report": "3a6b55a089d018878e8b904f8f19391f2e30b66c  colors.txt
477e8bc9954033392e432fd66f0d4278884bde38  cities.txt",
}
`;
